packages/client/src/components/Login.re

```diff
@@ -0,0 +1,125 @@
+module LoginMutationConfig = [%graphql
+  {|
+
+  mutation LoginMutation($username: RuleWrapper!, $password: String!) {
+    login(username: $username, password: $password) {
+      id
+      token
+      username
+    }
+  }
+
+|}
+];
+
+module LoginMutation = ReasonApolloHooks.Mutation.Make(LoginMutationConfig);
+
+type loginFormState = {
```

Comment:
I wonder if we could take benefit from variants here, so instead of working with booleans we could do:

```ocaml
type state = | Error(string) | Loading | Idle;
```

or:

```ocaml
type state = {
  error: option(string),
  loading: bool
};
```
we could then use:

```ocaml
state.error
->Belt.Option.map(error => error->React.string)
->Belt.Option.getWithDefault(React.null)
```